blueprints/incredible-squaring/src/eigenlayer.rs

```diff
@@ -0,0 +1,472 @@
+    // let my_msg_json = serde_json::to_string(&my_msg).unwrap();
+    // handle.send_message(my_msg_json).await;
+    //
+    // // 3. Receive gossiped results from peers
+    // let quorum_size = 1;
+    // let mut rx = handle.rx_from_inbound.lock().await;
+    // while let Some(message) = rx.recv().await {
+    //     let message: MyMessage = serde_json::from_str(&message).unwrap();
+    //     let public_key = ecdsa::Public::new_zero();
+    //     responses.insert(public_key, message);
+    //
+    //     if responses.len() >= quorum_size {
+    //         break;
+    //     }
+    // }
+    //
+    // // 4. Once we have a quorum, we calculate the non signers/participants and stakes
+    let non_signer_stakes_and_signature: NonSignerStakesAndSignature =
+        NonSignerStakesAndSignature {
+            nonSignerQuorumBitmapIndices: vec![], // Vec<u32>,
+            nonSignerPubkeys: vec![],             // Vec<<G1Point as SolType>::RustType>,
+            nonSignerStakeIndices: vec![],        // Vec<u32>,
+            quorumApks: vec![],                   // Vec<<G1Point as SolType>::RustType>,
+            apkG2: G2Point {
+                X: [U256::ZERO; 2], // <G2Point as SolType>::RustType,
+                Y: [U256::ZERO; 2], // <G2Point as SolType>::RustType,
+            },
+            sigma: G1Point {
+                X: U256::ZERO, // <G1Point as SolType>::RustType,
+                Y: U256::ZERO, // <G1Point as SolType>::RustType,
+            },
+            quorumApkIndices: vec![],  // Vec<u32>,
+            totalStakeIndices: vec![], // Vec<u32>,
+        };
+
+    // 5. We submit the full task response
+    Ok(respondToTaskCall {
+        task: Task {
+            numberToBeSquared: number_to_be_squared,
+            taskCreatedBlock: task_created_block,
+            quorumNumbers: quorum_numbers,
+            quorumThresholdPercentage: quorum_threshold_percentage,
+        },
+        taskResponse: TaskResponse {
+            referenceTaskIndex: task_created_block,
+            numberSquared: number_to_be_squared.saturating_pow(U256::from(2u32)),
+        },
+        nonSignerStakesAndSignature: non_signer_stakes_and_signature,
+    })
+}
+
+/// Converts the event to inputs.
+///
+/// Uses a tuple to represent the return type because
+/// the macro will index all values in the #[job] function
+/// and parse the return type by the index.
+pub fn convert_event_to_inputs(
+    event: IncredibleSquaringTaskManager::NewTaskCreated,
+) -> (U256, u32, Bytes, u32) {
+    let number_to_be_squared = event.task.numberToBeSquared;
+    let task_created_block = event.task.taskCreatedBlock;
+    let quorum_numbers = event.task.quorumNumbers;
+    let quorum_threshold_percentage = event.task.quorumThresholdPercentage;
+    (
+        number_to_be_squared,
+        task_created_block,
+        quorum_numbers,
+        quorum_threshold_percentage,
+    )
+}
+
+pub struct EigenlayerGadgetRunner<R: lock_api::RawRwLock> {
```

Comment:
I think this should be apart of the SDK. One idea would be to leverage a builder pattern or something that allows us to inject the custom AVS tooling into a generic runner so it can run without specifying all these boilerplate.